internal/services/sentinel/sentinel_data_connector_microsoft_threat_intelligence.go

```diff
@@ -0,0 +1,272 @@
+package sentinel
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/go-azure-sdk/resource-manager/operationalinsights/2022-10-01/workspaces"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/sentinel/parse"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/sentinel/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+	securityinsight "github.com/tombuildsstuff/kermit/sdk/securityinsights/2022-10-01-preview/securityinsights"
+)
+
+type DataConnectorMicrosoftThreatIntelligence struct{}
+
+type DataConnectorMicrosoftThreatIntelligenceModel struct {
+	Name                                    string `tfschema:"name"`
+	WorkspaceId                             string `tfschema:"log_analytics_workspace_id"`
+	TenantId                                string `tfschema:"tenant_id"`
+	BingSafetyPhishingUrlLookBackDate       string `tfschema:"bing_safety_phishing_url_lookback_date"`
+	MicrosoftEmergingThreatFeedLookBackDate string `tfschema:"microsoft_emerging_threat_feed_lookback_date"`
+}
+
+type DataConnectorMicrosoftThreatIntelligenceDataType struct {
+	Enabled      bool   `tfschema:"enabled"`
+	LookbackDate string `tfschema:"lookback_date"`
+}
+
+func (s DataConnectorMicrosoftThreatIntelligence) Arguments() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
```

Comment:
Is it limited with specific length and characters?

Reply:
I have not found limit on its name..